extensions/amp-list/0.1/amp-list.js

```diff
@@ -404,39 +409,30 @@ export class AmpList extends AMP.BaseElement {
           }
         });
       }
-    }, this.container_);
-
-    // This needs to be gated behind the experiment 'amp-list-resize'
-    if (resizeExperimentOn && !this.isSetToLayoutContainer_) {
-      this.changeToLayoutContainer_();
-      this.isSetToLayoutContainer_ = true;
-    }
+    });
   }
 
   /**
    * Converts the amp-list to de facto layout container.
    * @private
    */
   changeToLayoutContainer_() {
-    this.mutateElement(() => {
-      this.element.setAttribute('layout', 'container');
-      const sizer = childElementByTag(this.element, 'i-amphtml-sizer');
-      if (sizer) {
-        this.element.removeChild(sizer);
-      }
-      this.element.classList.remove(
-          'i-amphtml-layout-size-defined', 'i-amphtml-layout-responsive');
-      this.container_.classList.remove(
-          'i-amphtml-fill-content', 'i-amphtml-replaced-content');
-      // The overflow element is generally hidden with visibility hidden,
-      // but after changing to layout container, this causes an undesirable
-      // empty white space so we hide it with display none instead.
-      const overflowElement = this.getOverflowElement();
-      if (overflowElement) {
-        toggle(overflowElement, false);
-      }
-    });
-
+    this.element.setAttribute('layout', 'container');
+    const sizer = childElementByTag(this.element, 'i-amphtml-sizer');
+    if (sizer) {
+      this.element.removeChild(sizer);
+    }
+    this.element.classList.remove(
+        'i-amphtml-layout-size-defined', 'i-amphtml-layout-responsive');
```

Comment:
Removing `i-amphtml-layout-responsive` is right if `layout` was `responsive`, but what about the other [supported layouts](https://www.ampproject.org/docs/reference/components/amp-list)? Check out `getLayoutClass()` in layout.js. 

Looks like `nodisplay` is also supported but we probably shouldn't change that (make it visible suddenly). 

And while we're at it, should we be adding `i-amphtml-layout-container`? Basically undoing and reapplying `layout.js#applyStaticLayout()`. There are other things that need to be undone too for other layouts e.g. removing `width` and `height` styles.

Sorry to keep piling stuff on. 💩

Reply:
It seems like for layout fill and no-display, we shouldn't really do anything here.

Reply:
Another TODO: I realized that this implementation does not play nice with reset-on-refresh because the loader is also `visibility: hidden` instead of `display: none`, which will squash out the contents of the list. We may need to toggle it to false, and possibly add additional treatment to bring it back.

Reply:
On further testing, the issue with `reset-on-refresh` only applies to `flex-item` layout. It's possible that we could bail here and not handle it (i.e. revert to old approach). The main issue here is the loader taking up the full space of the list contents, and pushing the list contents below the list (so layout is completely weird). 

Options for solving this problem: 
1. Do nothing, revert to old approach. 
2. Toggle loader to false. (Bring it back on refresh?)

Reply:
Actually it's not possible to override this because it has `display: block !important`, so maybe what we want to do is rethink the approach to reusable loaders. This seems to be a smaller edge case and possibly worth separating into a different PR since it may also touch loader logic. I've reverted `flex-item` layout to use the original approach as a stop gap for this PR.

Reply:
sgtm, `flex-item` is not commonly used and we can deal with it later.